dataplane/fragmentation.cpp

```diff
@@ -1,19 +1,39 @@
 #include "fragmentation.h"
 #include "common.h"
 #include "controlplane.h"
-#include "dataplane.h"
+#include "metadata.h"
 
-fragmentation_t::fragmentation_t(cControlPlane* controlPlane,
-                                 cDataPlane* dataPlane) :
-        controlPlane(controlPlane),
-        dataPlane(dataPlane)
+namespace fragmentation
 {
-	memset(&stats, 0, sizeof(stats));
+
+Fragmentation::Fragmentation(OnCollected callback) :
+        callback_{callback}
+{}
```

Comment:
maybe memset `stats_` here too?

Reply:
Actually we can just use value initialization from `stats_` field: `common::fragmentation::stats_t stats_{};`
`stats_t`does not have any default arguments and will zero itself